test/sim/abilities/ripen.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe("Ripen", function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it("should double healing from berries", function () {
+		battle = common.createBattle();
+		battle.setPlayer('p1', {team: [
+			{species: 'Appletun', ability: 'ripen', item: 'sitrusberry', moves: ['growth'], ivs: {hp: 0}},
+		]});
+		battle.setPlayer('p2', {team: [
+			{species: 'Skwovet', ability: 'gluttony', moves: ['superfang']},
+		]});
+		battle.makeChoices();
+		assert.fullHP(battle.p1.active[0]); // Damaged by 50% + healed 50% = 100%
```

Comment:
Travis errors here, error: "AssertionError [ERR_ASSERTION]: Expected p1a: Appletun to be fully healed, not at 329/330." 
Will pull and check out later. Potentially off by one error?

Reply:
I confirmed this with a researcher, the 329/330 is actually correct behavior. Ripen does indeed calculate `floor(hp / 4) * 2` as opposed to `floor(hp / 4 * 2)`. I will fix this when I get the chance, just been busy lately